src/tqec/templates/library/spatial.py

```diff
@@ -0,0 +1,430 @@
+from tqec.templates.indices.qubit import (
+    QubitHorizontalBorders,
+    QubitSpatialJunctionTemplate,
+    QubitVerticalBorders,
+)
+from tqec.templates.rpng import RPNGTemplate
+
+
+def get_spatial_junction_qubit_template(
+    external_stabilizers: Literal["x", "z"],
+    arms: JunctionArms,
+    reset: ResetBasis | None = None,
+    measurement: MeasurementBasis | None = None,
+) -> RPNGTemplate:
+    """Implementation of a logical qubit performing a spatial junction.
+
+    This function returns a RPNGTemplate instance representing a logical qubit
+    that is touched by 2 or more spatial junctions. The returned template is
+    carefully crafted to avoid hook errors damaging the logical distance.
+
+    Note:
+        this function does not enforce anything on the input values. As such, it
+        is possible to generate a description of a round that will both reset and
+        measure the data-qubits.
+
+    Warning:
+        By convention, this function does not populate the plaquettes on the
+        boundaries where an arm (i.e. spatial junction) is present **BUT** do
+        populate the corners (that are part of the boundaries, so this is an
+        exception to the first part of the sentence).
+
+        The rationale behind that convention is that the logical qubit at the
+        center of the spatial junction is completely aware of all the spatial
+        junctions that should be implemented whereas each spatial junction in
+        isolation does not know if the others spatial junctions are present or
+        not. That means that corners, whose plaquette depends on the presence or
+        absence of the two spatial junctions it belongs to, require information
+        that is given to this function, but not to the junction generation
+        function.
+
+        Junctions should follow that convention and should not replace the
+        plaquette descriptions on the corners (i.e., not include an explicit
+        mapping, even to the empty plaquette, from the index of the corner to
+        a plaquette).
+
+    Warning:
+        Corners are populated with plaquette that do not contain resets or
+        measurements on the data-qubits. It is up to the junction implementation
+        to make sure that a reset or a measurement is present on the qubit(s)
+        shared with the corner, if needed.
+
+        The rationale behind that convention is that the information about
+        whether or not the internal data-qubits of a junction should be reset
+        or measured is given to the junction-generation function, and not to
+        the qubit-generation function (this one). It is also important to note
+        that the information we need here is not given by ``reset`` or
+        ``measurement``: the logical qubit at the center of the spatial junction
+        can already be initialised (and so have an incoming temporal junction
+        from below), which means ``reset is None`` and ``measurement is None``,
```

Comment:
A spatial junction should not have temporal pipes connecting to them. Otherwise, the boundary conditions will not match.

Reply:
I just removed that assumption on the spatial junction (i.e., the logical qubit). It might also simplify the arms implementation, but I have not done it yet. If you see a potential improvement in your review, please let me know.